### DISCUSSION TITLE
How to set up a channel on Mirth where the message can be deleted after receiving an ACK

### DISCUSSION THREAD

**Discussion by trilochanmukkala** [nextgenhealthcare/connect#6326] [Q&A] [answered]
I'm using Mirth 4.5.2 and set up a channel to use database reader as source and TCP sender as destination. TCP sender takes the message from database, sends the messages to a TCP listener port (not part of Mirth) which processes the HL7 messages and sends back ACK. How can I set up the channel such that the messages from the database are deleted once ACK is received from the TCP listener

**Comment by jonbartels** [comment]
https://docs.nextgen.com/bundle/Mirth_User_Guide_4_5_0/page/connect/connect/topics/c_Message_Storage_Settings_connect_ug.html

You want "Remove Content on Completion".

**Reply by trilochanmukkala** [reply]
But the source is not a table created by Mirth, it's a custom table added to the database
will it work?

**Reply by pacmano1** [reply, marked as answer]
then delete the row from the table after you do whatever you need to do with it.

You can do that right from the source tab on the bottom of the database reader.   Some training and reading the user manual would be a good idea.

https://docs.nextgen.com/bundle/Mirth_User_Guide_4_5_0/page/connect/connect/topics/c_Database_Reader_connect_ug.html

![image](https://github.com/user-attachments/assets/944f0dcc-8897-467a-92b8-e0a580dee46f)

A good training course:  https://www.udemy.com/course/learning-mirth-connect/
